inc/classes/wp-rocket-uninstall.php

```diff
@@ -0,0 +1,212 @@
+<?php
+
+/**
+ * Manages the deletion of WP Rocket data and files on uninstall
+ */
+class WP_Rocket_Uninstall {
+	/**
+	 * Path to the cache folder
+	 *
+	 * @var string
+	 */
+	private $cache_path;
+
+	/**
+	 * Path to the config folder
+	 *
+	 * @var string
+	 */
+	private $config_path;
+
+	/**
+	 * WP Rocket options
+	 *
+	 * @var array
+	 */
+	private $options = [
+		'wp_rocket_settings',
+		'rocket_analytics_notice_displayed',
+		'rocketcdn_user_token',
+		'rocketcdn_process',
+	];
+
+	/**
+	 * WP Rocket transients
+	 *
+	 * @var array
+	 */
+	private $transients = [
+		'wp_rocket_customer_data',
+		'rocket_notice_missing_tags',
+		'rocket_clear_cache',
+		'rocket_check_key_errors',
+		'rocket_send_analytics_data',
+		'rocket_critical_css_generation_process_running',
+		'rocket_critical_css_generation_process_complete',
+		'rocket_critical_css_generation_triggered',
+		'rocketcdn_status',
+		'rocketcdn_pricing',
+		'rocketcdn_purge_cache_response',
+		'rocket_cloudflare_ips',
+		'rocket_cloudflare_is_api_keys_valid',
+		'rocket_preload_triggered',
+		'rocket_preload_complete',
+		'rocket_preload_complete_time',
+		'rocket_preload_errors',
+		'rocket_database_optimization_process',
+		'rocket_database_optimization_process_complete',
+	];
+
+	/**
+	 * WP Rocket scheduled events
+	 *
+	 * @var array
+	 */
+	private $events = [
+		'rocket_purge_time_event',
+		'rocket_database_optimization_time_event',
+		'rocket_google_tracking_cache_update',
+		'rocket_facebook_tracking_cache_update',
+		'rocket_cache_dir_size_check',
+		'rocketcdn_check_subscription_status_event',
+		'rocket_cron_deactivate_cloudflare_devmode',
+	];
+
+	/**
+	 * WP Rocket cache directories
+	 *
+	 * @var array
+	 */
+	private $cache_dirs = [
+		'wp-rocket',
+		'min',
+		'busting',
+		'critical-css',
+	];
+
+	/**
+	 * Constructor
+	 *
+	 * @param string $cache_path  Path to the cache folder.
+	 * @param string $config_path Path to the config folder.
+	 */
+	public function __construct( $cache_path, $config_path ) {
+		$this->cache_path  = trailingslashit( $cache_path );
+		$this->config_path = $config_path;
+	}
+
+	/**
+	 * Deletes all plugin data and files on uninstall
+	 *
+	 * @since 3.5.2
+	 * @author Remy Perona
+	 *
+	 * @return void
+	 */
+	public function uninstall() {
+		$this->delete_plugin_data();
+		$this->delete_cache_files();
+		$this->delete_config_files();
+	}
+
+	/**
+	 * Deletes WP Rocket options, transients and events
+	 *
+	 * @since 3.5.2
+	 * @author Remy Perona
+	 *
+	 * @return void
+	 */
+	private function delete_plugin_data() {
+		foreach ( $this->transients as $transient ) {
+			delete_transient( $transient );
+		}
```

Comment:
What about using `array_map()`?
```suggestion
		array_map( 'delete_transient', $this->transients );
```